dashboard/app: TestAIJobParallelPoll failure

Observed on https://github.com/google/syzkaller/actions/runs/24343633906/job/71078587141?pr=7053

```
2026-04-13T12:40:31.3619454Z --- FAIL: TestAIJobParallelPoll (4.44s)
2026-04-13T12:40:31.3623032Z     util_test.go:652: : API(upload_build): &dashapi.Build{Manager:"manager1", ID:"build1", OS:"linux", Arch:"amd64", VMArch:"amd64", SyzkallerCommit:"syzkaller_commit1", SyzkallerCommitDate:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), CompilerID:"compiler1", KernelRepo:"repo1", KernelBranch:"branch1", KernelCommit:"1111111111111111111111111111111111111111", KernelCommitTitle:"kernel_commit_title1", KernelCommitDate:time.Date(1, time.February, 3, 4, 5, 6, 0, time.UTC), KernelConfig:[]uint8{0x63, 0x6f, 0x6e, 0x66, 0x69, 0x67, 0x31}, Commits:[]string(nil), FixCommits:[]dashapi.Commit(nil), Assets:[]dashapi.NewAsset(nil)}
2026-04-13T12:40:31.3634299Z ##[error]: API(upload_build): &dashapi.Build{Manager:"manager1", ID:"build1", OS:"linux", Arch:"amd64", VMArch:"amd64", SyzkallerCommit:"syzkaller_commit1", SyzkallerCommitDate:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), CompilerID:"compiler1", KernelRepo:"repo1", KernelBranch:"branch1", KernelCommit:"1111111111111111111111111111111111111111", KernelCommitTitle:"kernel_commit_title1", KernelCommitDate:time.Date(1, time.February, 3, 4, 5, 6, 0, time.UTC), KernelConfig:[]uint8{0x63, 0x6f, 0x6e, 0x66, 0x69, 0x67, 0x31}, Commits:[]string(nil), FixCommits:[]dashapi.Commit(nil), Assets:[]dashapi.NewAsset(nil)}
2026-04-13T12:40:31.3639237Z     util_test.go:652: : API(upload_build): REPLY: <nil>
2026-04-13T12:40:31.3643291Z ##[error]: API(upload_build): REPLY: <nil>
2026-04-13T12:40:31.3648147Z     util_test.go:652: : API(report_crash): &dashapi.Crash{BuildID:"build1", Title:"KCSAN: data-race in foo / bar", AltTitles:[]string(nil), Corrupted:false, Suppressed:false, Maintainers:[]string(nil), Recipients:dashapi.Recipients(nil), Log:[]uint8{0x6c, 0x6f, 0x67, 0x31}, Flags:0, Report:[]uint8{0x72, 0x65, 0x70, 0x6f, 0x72, 0x74, 0x31}, MachineInfo:[]uint8{0x6d, 0x61, 0x63, 0x68, 0x69, 0x6e, 0x65, 0x20, 0x69, 0x6e, 0x66, 0x6f, 0x20, 0x31}, Assets:[]dashapi.NewAsset(nil), GuiltyFiles:[]string(nil), ReproOpts:[]uint8(nil), ReproSyz:[]uint8(nil), ReproC:[]uint8(nil), ReproLog:[]uint8(nil), OriginalTitle:""}
2026-04-13T12:40:31.3658485Z ##[error]: API(report_crash): &dashapi.Crash{BuildID:"build1", Title:"KCSAN: data-race in foo / bar", AltTitles:[]string(nil), Corrupted:false, Suppressed:false, Maintainers:[]string(nil), Recipients:dashapi.Recipients(nil), Log:[]uint8{0x6c, 0x6f, 0x67, 0x31}, Flags:0, Report:[]uint8{0x72, 0x65, 0x70, 0x6f, 0x72, 0x74, 0x31}, MachineInfo:[]uint8{0x6d, 0x61, 0x63, 0x68, 0x69, 0x6e, 0x65, 0x20, 0x69, 0x6e, 0x66, 0x6f, 0x20, 0x31}, Assets:[]dashapi.NewAsset(nil), GuiltyFiles:[]string(nil), ReproOpts:[]uint8(nil), ReproSyz:[]uint8(nil), ReproC:[]uint8(nil), ReproLog:[]uint8(nil), OriginalTitle:""}
2026-04-13T12:40:31.3662816Z     util_test.go:652: : API(report_crash): REPLY: &dashapi.ReportCrashResp{NeedRepro:true}
2026-04-13T12:40:31.3664615Z ##[error]: API(report_crash): REPLY: &dashapi.ReportCrashResp{NeedRepro:true}
2026-04-13T12:40:31.3666070Z     util_test.go:489: GET: /cron/email_poll
2026-04-13T12:40:31.3667529Z ##[error]GET: /cron/email_poll
2026-04-13T12:40:31.3668578Z     util_test.go:508: REPLY: 200
2026-04-13T12:40:31.3669598Z ##[error]REPLY: 200
2026-04-13T12:40:31.3670558Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3672197Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3674189Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-3", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3678147Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-3", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3680064Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3681139Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3682951Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-7", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3686733Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-7", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3688683Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3689770Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3691730Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-0", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3695548Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-0", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3697799Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3698873Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3700735Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-9", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3704565Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-9", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3707121Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3708254Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3710201Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-2", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3714226Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-2", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3716873Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3717962Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3719856Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-5", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3723515Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-5", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3725509Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3726974Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3728968Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-6", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3732613Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-6", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3734806Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3735847Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3738032Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-12", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3741542Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-12", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3743413Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3744380Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3746201Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-4", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3749982Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-4", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3751996Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3753086Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3755041Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-11", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3759085Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-11", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3761063Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3762189Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3764328Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-8", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3768284Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-8", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3770259Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3771206Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3773054Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-10", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3777077Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-10", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3779230Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3780311Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3782254Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-13", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3785923Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-13", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3788216Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3789289Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3791379Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-14", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3795136Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-14", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3797589Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3798673Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3800646Z         errgroup.go:93: API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-1", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3804360Z ##[error]API(ai_job_poll): &dashapi.AIJobPollReq{AgentName:"agent-1", CodeRevision:"unknown", Workflows:[]dashapi.AIWorkflow{dashapi.AIWorkflow{Type:"assessment-kcsan", Name:"kcsan-job"}}}
2026-04-13T12:40:31.3806744Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3807749Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3809170Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.3811260Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.3812702Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3813723Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3815093Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.3817644Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.3819103Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3820145Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3821554Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.3823669Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.3825023Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3825851Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3828872Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"c810b4dd-4ce8-4bcf-8875-6c2265f37394", Workflow:"kcsan-job", Args:map[string]interface {}{"BugTitle":"KCSAN: data-race in foo / bar", "CrashLog":"log1", "CrashReport":"report1", "KernelCommit":"1111111111111111111111111111111111111111", "KernelConfig":"config1", "KernelRepo":"repo1", "ReproOpts":"", "SyzkallerCommit":"syzkaller_commit1"}}
2026-04-13T12:40:31.3835258Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"c810b4dd-4ce8-4bcf-8875-6c2265f37394", Workflow:"kcsan-job", Args:map[string]interface {}{"BugTitle":"KCSAN: data-race in foo / bar", "CrashLog":"log1", "CrashReport":"report1", "KernelCommit":"1111111111111111111111111111111111111111", "KernelConfig":"config1", "KernelRepo":"repo1", "ReproOpts":"", "SyzkallerCommit":"syzkaller_commit1"}}
2026-04-13T12:40:31.3838566Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3839783Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3841106Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.3843001Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.3844217Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3845080Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3847976Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"c810b4dd-4ce8-4bcf-8875-6c2265f37394", Workflow:"kcsan-job", Args:map[string]interface {}{"BugTitle":"KCSAN: data-race in foo / bar", "CrashLog":"log1", "CrashReport":"report1", "KernelCommit":"1111111111111111111111111111111111111111", "KernelConfig":"config1", "KernelRepo":"repo1", "ReproOpts":"", "SyzkallerCommit":"syzkaller_commit1"}}
2026-04-13T12:40:31.3854327Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"c810b4dd-4ce8-4bcf-8875-6c2265f37394", Workflow:"kcsan-job", Args:map[string]interface {}{"BugTitle":"KCSAN: data-race in foo / bar", "CrashLog":"log1", "CrashReport":"report1", "KernelCommit":"1111111111111111111111111111111111111111", "KernelConfig":"config1", "KernelRepo":"repo1", "ReproOpts":"", "SyzkallerCommit":"syzkaller_commit1"}}
2026-04-13T12:40:31.3857384Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3858237Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3859622Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.3861886Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.3863393Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3864417Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.3865822Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.3868322Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.3869739Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.3870709Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.4306696Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4308241Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4309122Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.4309780Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.4310587Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4311800Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4312620Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.4313200Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.4314003Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4315215Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4315994Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.4317092Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.4317884Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4319074Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4319840Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.4320396Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.4321751Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4323829Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4325117Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.4326069Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.4327567Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4328822Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4329608Z     util_test.go:652: asm_amd64.s:1771
2026-04-13T12:40:31.4330180Z ##[error]asm_amd64.s:1771
2026-04-13T12:40:31.4330978Z         errgroup.go:93: API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4332217Z ##[error]API(ai_job_poll): REPLY: &dashapi.AIJobPollResp{ID:"", Workflow:"", Args:map[string]interface {}(nil)}
2026-04-13T12:40:31.4332980Z     ai_test.go:634: 
2026-04-13T12:40:31.4333422Z         	Error Trace:	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/dashboard/app/ai_test.go:634
2026-04-13T12:40:31.4333933Z         	Error:      	Not equal: 
2026-04-13T12:40:31.4334146Z         	            	expected: 1
2026-04-13T12:40:31.4334352Z         	            	actual  : 2
2026-04-13T12:40:31.4334571Z         	Test:       	TestAIJobParallelPoll
```

--- a/dashboard/app/aidb/crud.go
+++ b/dashboard/app/aidb/crud.go
@@ -157,6 +157,9 @@ func StartJob(ctx context.Context, req *dashapi.AIJobPollReq) (*Job, error) {
 	}
 	var job *Job
 	_, err = client.ReadWriteTransaction(ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+		// Reset job from previous transaction runs, otherwise we might return stale
+		// result if the job has been taken by a concurent thread (the len(jobs) == 0 case).
+		job = nil
 		iter := txn.Query(ctx, spanner.Statement{
 			SQL: selectJobs() + `WHERE Workflow IN UNNEST(@workflows)
 					AND Started IS NULL
@@ -192,6 +195,8 @@ func NextStaleJob(ctx context.Context, req *dashapi.AIJobPollReq) (*Job, error) 
 	cutoff := TimeNow(ctx).Add(-8 * time.Hour)
 	var job *Job
 	_, err = client.ReadWriteTransaction(ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+		// Avoid reusing the entities from previous transaction commit attempts.
+		job = nil
 		var jobs []*Job
 
 		// First, check if the requesting agent has any unfinished jobs (implying a restart).

